libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -72,11 +71,9 @@ internal class RouteRefreshController(
         }
     }
 
-    private fun refreshRoute() {
-        val route = tripSession.route
-            ?.takeIf { it.routeOptions()?.enableRefresh() == true }
-            ?.takeIf { it.isUuidValidForRefresh() }
-        if (route != null) {
+    private fun refreshRoute(route: DirectionsRoute) {
+        val isValid = route.routeOptions()?.enableRefresh() == true && route.isUuidValidForRefresh()
```

Comment:
I'm a little on the fence of having it come from the external caller. The tripSession is still needed with route progress

Reply:
IMO we should just call routeRefreshController.restart() without the route

RouteRefreshController depends on directionsSession so it can access the primary route on its own.

Also, that reduces the tests you need to add to the overpowered MapboxNavigation.